### PR TITLE
Allow Notifier deprecations to be registered

### PR DESCRIPTION
**Pull request by drbyte**
Adds the following methods to Notifier and Observer base traits:
`registerDeprecatedEvent($old, $new)`
`registerObserverAlias($old, $new)`

### CODE REVIEW

File: includes/classes/traits/NotifierManager.php [hunk 1]
```diff
@@ -126,16 +147,26 @@ protected function logNotifier($eventID, $param1, $param2, $param3, $param4, $pa
         error_log($zcDate->output("%Y-%m-%d %H:%M:%S") . ' [main_page=' . $main_page . '] ' . $eventID . $output . "\n", 3, $file);
     }
 
-    private function eventIdHasAlias($eventId)
+    private function eventIdHasAlias($eventId): bool
     {
         if (array_key_exists($eventId, $this->observerAliases)) {
             return true;
         }
+
```
Comment (lat9) [inline includes/classes/traits/NotifierManager.php:155]: Why not just `return array_key_exists($eventId, $this->observerAliases);` directly?

**Review by lat9 — COMMENTED** [review]
Looks good, just one small (and optional) refinement.

File: includes/classes/traits/NotifierManager.php (hunk 1 above)
Comment (drbyte) [reply includes/classes/traits/NotifierManager.php:155]: Ya, I was thinking the same. It was an existing method so I didn't change it. But I agree. Will update.

Comment (drbyte) [reply includes/classes/traits/NotifierManager.php]: Updated.

**Pull request merged**